torch/csrc/distributed/rpc/init.cpp

```diff
@@ -513,11 +493,27 @@ If the future completes with an error, an exception is thrown.
          std::string& pickledPythonUDF,
          std::vector<torch::Tensor>& tensors,
          const float rpcTimeoutSeconds) {
-        DCHECK(!PyGILState_Check());
-        return pyRpcPythonUdf(
-            dst, pickledPythonUDF, tensors, rpcTimeoutSeconds);
+        return std::make_shared<torch::jit::PythonFutureWrapper>(
+            pyRpcPythonUdf(dst, pickledPythonUDF, tensors, rpcTimeoutSeconds),
+            [](const py::object& value) {
+              py::gil_scoped_release release;
+              auto& pythonRpcHandler = PythonRpcHandler::getInstance();
```

Comment:
should we also detect if `PythonRpcHandler` has been cleaned up?

Reply:
Good idea. Preferring doing it in a next PR. It should also be be carefully designed. And I guess it will just be a best-effort bool flag for performance reason.